diff/fd/crosslaplacian.go

```diff
@@ -0,0 +1,190 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package fd
+
+import (
+	"math"
+	"sync"
+)
+
+// CrossLaplacian computes a Laplacian-like quantity for a function of two vectors
+// at the locations x and y.
+// It computes
+//  ∇_y · ∇_x f(x,y) = \sum_i ∂^2 f(x,y)/∂x_i ∂y_i
```

Comment:
Sorry, I take it back, the length restriction can't be dropped.

If we take the hessian as

```
[  ∂^2 f(x,y)/dx^2    ∂^2 f(x,y)/dxdy
   ∂^2 f(x,y)/dxdx    ∂^2 f(x,y)/dy^2 ]
```
which would be the blocks of the Hessian if we concatenate `x` and `y` together. This computes the diagonal of `∂^2 f(x,y) /dxdx`, while your formula computes the sum of all of the elements in that submatrix. I could imagine we would also want a `CrossHessian` which returns the full matrix instead of its Laplacian. For the Laplacian though, the vector lengths have to be equal.